### PR TITLE
ANY -> Any -> T where T

### PR DESCRIPTION
**Pull request by ararslan**

### CODE REVIEW

**Comment by ararslan** [comment]
@JeffBezanson @andreasnoack I have a feeling this failure (which I'm realizing was in the other PR that I hastily merged) seems like it may be related to not using `ANY`?
```
ERROR: LoadError: return type Array{Float64,2} does not match inferred return type Any
Stacktrace:
 [1] macro expansion at /home/travis/.julia/v0.7/FFTW/test/runtests.jl:335 [inlined]
 [2] anonymous at ./<missing>:?
 [3] include_relative(::Module, ::String) at ./loading.jl:464
 [4] include(::Module, ::String) at ./sysimg.jl:14
 [5] process_options(::Base.JLOptions) at ./client.jl:315
 [6] _start() at ./client.jl:383
while loading /home/travis/.julia/v0.7/FFTW/test/runtests.jl, in expression starting on line 331
```

File: src/fft.jl [hunk 1]
```diff
@@ -541,10 +541,10 @@ for (Tr,Tc,fftw,lib) in ((:Float64,:Complex128,"fftw",libfftw),
         return rFFTWPlan{$Tc,$BACKWARD,inplace,N}(plan, flags, R, X, Y)
     end
 
-    @eval function (::Type{r2rFFTWPlan{$Tr,Any,inplace,N}})(X::StridedArray{$Tr,N},
-                                                            Y::StridedArray{$Tr,N},
-                                                            region, kinds, flags::Integer,
-                                                            timelimit::Real) where {inplace,N}
+    @eval function (::Type{r2rFFTWPlan{$Tr,T,inplace,N}} where T)(X::StridedArray{$Tr,N},
```
Comment (musm) [inline src/fft.jl]: This looks a little odd, why not place this with the type parameters on line 547? Likewise below.

**Comment by JeffBezanson** [comment]
You could try moving `where T` outside, putting `T` with the rest of the method parameters `{inplace,N}`. See if that helps.

**Comment by ararslan** [comment]
Doesn't seem to have made a difference, though that's assuming I've correctly identified the current inference failure as having to do with the switch from `ANY`.

**Comment by mschauer** [comment]
Travis pr passes now.

**Comment by ararslan** [comment]
This is actually no longer necessary, as I incorporated these changes in a different PR.

**Pull request closed without merging**